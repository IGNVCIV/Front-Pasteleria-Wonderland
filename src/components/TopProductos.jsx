import { useEffect, useState } from "react";

function TopProductosCriticos() {
  const [productos, setProductos] = useState([]);

  useEffect(() => {
    const catalogo = JSON.parse(localStorage.getItem("catalogoProductos")) || [];

    const catalogoConVentas = catalogo.map((prod) => ({
      ...prod,
      ventasSemanales: Math.floor(Math.random() * (60 - 5 + 1)) + 5,
    }));

    const top10 = catalogoConVentas.slice(0, 10);

    setProductos(top10);
  }, []);

  return (
    <div className="container mt-4">
      <h2 className="text-center mb-4">
        <span style={{ display: "inline-block", marginRight: "8px" }}>üç∞</span>
        Top 10 Productos Cr√≠ticos
      </h2>

      <div
        className="d-flex flex-row overflow-auto p-3 gap-3"
        style={{
          border: "1px solid #ddd",
          borderRadius: "12px",
          backgroundColor: "#fff",
          boxShadow: "0 2px 6px rgba(0,0,0,0.1)",
          scrollBehavior: "smooth",
        }}
      >
        {productos.length > 0 ? (
          productos.map((prod, index) => (
            <div
              key={prod.id}
              className="p-3 text-center"
              style={{
                minWidth: "250px",
                border: "1px solid #e0e0e0",
                borderRadius: "10px",
                backgroundColor: "#fdfaf9",
                transition: "transform 0.2s ease",
              }}
              onMouseEnter={(e) => (e.currentTarget.style.transform = "scale(1.05)")}
              onMouseLeave={(e) => (e.currentTarget.style.transform = "scale(1)")}
            >
              <h5 className="mb-2 text-primary">#{index + 1}</h5>
              <img
                src={prod.imagen}
                alt={prod.nombre}
                className="d-block mx-auto mb-2"
                style={{ width: "100px", height: "100px", objectFit: "cover" }}
              />
              <p className="mb-1">
                <strong>ID:</strong> {prod.id}
              </p>
              <p className="mb-1">
                <strong>Nombre:</strong> {prod.nombre}
              </p>
              <p className="mb-0">
                <strong>Ventas semanales:</strong> {prod.ventasSemanales} unidades
              </p>
            </div>
          ))
        ) : (
          <p className="text-center w-100">No hay productos disponibles.</p>
        )}
      </div>
    </div>
  );
}

export default TopProductosCriticos;
