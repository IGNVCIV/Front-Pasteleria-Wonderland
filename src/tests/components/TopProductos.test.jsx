import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { describe, it, expect, beforeEach, afterEach } from "vitest";
import TopProductos from "../../components/TopProductos";

describe("TopProductos Component", () => {
  const catalogoMock = Array.from({ length: 12 }, (_, i) => ({
    id: i + 1,
    nombre: `Producto ${i + 1}`,
    imagen: `https://example.com/prod${i + 1}.jpg`,
  }));

  beforeEach(() => {
    localStorage.clear();
  });

  afterEach(() => {
    localStorage.clear();
  });

  it("muestra mensaje cuando no hay productos", () => {
    render(<TopProductos />);
    expect(screen.getByText(/No hay productos disponibles/i)).toBeInTheDocument();
  });

  it("renderiza correctamente los productos del localStorage", () => {
    localStorage.setItem("catalogoProductos", JSON.stringify(catalogoMock));
    render(<TopProductos />);

    for (let i = 0; i < 10; i++) {
      expect(screen.getByText(`Producto ${i + 1}`)).toBeInTheDocument();
      expect(screen.getByText(`#${i + 1}`)).toBeInTheDocument();
    }

    expect(screen.queryByText("Producto 11")).not.toBeInTheDocument();
    expect(screen.queryByText("Producto 12")).not.toBeInTheDocument();
  });
});
