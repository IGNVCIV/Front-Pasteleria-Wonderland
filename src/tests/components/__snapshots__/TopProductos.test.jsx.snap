// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TopProductos Component > coincide con el snapshot 1`] = `
<div>
  <div
    class="container mt-4"
  >
    <h2
      class="text-center mb-4"
    >
      <span
        style="display: inline-block; margin-right: 8px;"
      >
        üç∞
      </span>
      Top 10 Productos Cr√≠ticos
    </h2>
    <div
      class="d-flex flex-row overflow-auto p-3 gap-3"
      style="border-radius: 12px; background-color: rgb(255, 255, 255); box-shadow: 0 2px 6px rgba(0,0,0,0.1); scroll-behavior: smooth;"
    >
      <div
        class="p-3 text-center"
        style="min-width: 250px; border-radius: 10px; background-color: rgb(253, 250, 249); transition: transform 0.2s ease;"
      >
        <h5
          class="mb-2 text-primary"
        >
          #
          1
        </h5>
        <img
          alt="Producto 1"
          class="d-block mx-auto mb-2"
          src="https://example.com/prod1.jpg"
          style="width: 100px; height: 100px; object-fit: cover;"
        />
        <p
          class="mb-1"
        >
          <strong>
            ID:
          </strong>
           
          1
        </p>
        <p
          class="mb-1"
        >
          <strong>
            Nombre:
          </strong>
           
          Producto 1
        </p>
        <p
          class="mb-0"
        >
          <strong>
            Ventas semanales:
          </strong>
           
          31
           unidades
        </p>
      </div>
      <div
        class="p-3 text-center"
        style="min-width: 250px; border-radius: 10px; background-color: rgb(253, 250, 249); transition: transform 0.2s ease;"
      >
        <h5
          class="mb-2 text-primary"
        >
          #
          2
        </h5>
        <img
          alt="Producto 2"
          class="d-block mx-auto mb-2"
          src="https://example.com/prod2.jpg"
          style="width: 100px; height: 100px; object-fit: cover;"
        />
        <p
          class="mb-1"
        >
          <strong>
            ID:
          </strong>
           
          2
        </p>
        <p
          class="mb-1"
        >
          <strong>
            Nombre:
          </strong>
           
          Producto 2
        </p>
        <p
          class="mb-0"
        >
          <strong>
            Ventas semanales:
          </strong>
           
          7
           unidades
        </p>
      </div>
      <div
        class="p-3 text-center"
        style="min-width: 250px; border-radius: 10px; background-color: rgb(253, 250, 249); transition: transform 0.2s ease;"
      >
        <h5
          class="mb-2 text-primary"
        >
          #
          3
        </h5>
        <img
          alt="Producto 3"
          class="d-block mx-auto mb-2"
          src="https://example.com/prod3.jpg"
          style="width: 100px; height: 100px; object-fit: cover;"
        />
        <p
          class="mb-1"
        >
          <strong>
            ID:
          </strong>
           
          3
        </p>
        <p
          class="mb-1"
        >
          <strong>
            Nombre:
          </strong>
           
          Producto 3
        </p>
        <p
          class="mb-0"
        >
          <strong>
            Ventas semanales:
          </strong>
           
          37
           unidades
        </p>
      </div>
      <div
        class="p-3 text-center"
        style="min-width: 250px; border-radius: 10px; background-color: rgb(253, 250, 249); transition: transform 0.2s ease;"
      >
        <h5
          class="mb-2 text-primary"
        >
          #
          4
        </h5>
        <img
          alt="Producto 4"
          class="d-block mx-auto mb-2"
          src="https://example.com/prod4.jpg"
          style="width: 100px; height: 100px; object-fit: cover;"
        />
        <p
          class="mb-1"
        >
          <strong>
            ID:
          </strong>
           
          4
        </p>
        <p
          class="mb-1"
        >
          <strong>
            Nombre:
          </strong>
           
          Producto 4
        </p>
        <p
          class="mb-0"
        >
          <strong>
            Ventas semanales:
          </strong>
           
          43
           unidades
        </p>
      </div>
      <div
        class="p-3 text-center"
        style="min-width: 250px; border-radius: 10px; background-color: rgb(253, 250, 249); transition: transform 0.2s ease;"
      >
        <h5
          class="mb-2 text-primary"
        >
          #
          5
        </h5>
        <img
          alt="Producto 5"
          class="d-block mx-auto mb-2"
          src="https://example.com/prod5.jpg"
          style="width: 100px; height: 100px; object-fit: cover;"
        />
        <p
          class="mb-1"
        >
          <strong>
            ID:
          </strong>
           
          5
        </p>
        <p
          class="mb-1"
        >
          <strong>
            Nombre:
          </strong>
           
          Producto 5
        </p>
        <p
          class="mb-0"
        >
          <strong>
            Ventas semanales:
          </strong>
           
          24
           unidades
        </p>
      </div>
      <div
        class="p-3 text-center"
        style="min-width: 250px; border-radius: 10px; background-color: rgb(253, 250, 249); transition: transform 0.2s ease;"
      >
        <h5
          class="mb-2 text-primary"
        >
          #
          6
        </h5>
        <img
          alt="Producto 6"
          class="d-block mx-auto mb-2"
          src="https://example.com/prod6.jpg"
          style="width: 100px; height: 100px; object-fit: cover;"
        />
        <p
          class="mb-1"
        >
          <strong>
            ID:
          </strong>
           
          6
        </p>
        <p
          class="mb-1"
        >
          <strong>
            Nombre:
          </strong>
           
          Producto 6
        </p>
        <p
          class="mb-0"
        >
          <strong>
            Ventas semanales:
          </strong>
           
          38
           unidades
        </p>
      </div>
      <div
        class="p-3 text-center"
        style="min-width: 250px; border-radius: 10px; background-color: rgb(253, 250, 249); transition: transform 0.2s ease;"
      >
        <h5
          class="mb-2 text-primary"
        >
          #
          7
        </h5>
        <img
          alt="Producto 7"
          class="d-block mx-auto mb-2"
          src="https://example.com/prod7.jpg"
          style="width: 100px; height: 100px; object-fit: cover;"
        />
        <p
          class="mb-1"
        >
          <strong>
            ID:
          </strong>
           
          7
        </p>
        <p
          class="mb-1"
        >
          <strong>
            Nombre:
          </strong>
           
          Producto 7
        </p>
        <p
          class="mb-0"
        >
          <strong>
            Ventas semanales:
          </strong>
           
          7
           unidades
        </p>
      </div>
      <div
        class="p-3 text-center"
        style="min-width: 250px; border-radius: 10px; background-color: rgb(253, 250, 249); transition: transform 0.2s ease;"
      >
        <h5
          class="mb-2 text-primary"
        >
          #
          8
        </h5>
        <img
          alt="Producto 8"
          class="d-block mx-auto mb-2"
          src="https://example.com/prod8.jpg"
          style="width: 100px; height: 100px; object-fit: cover;"
        />
        <p
          class="mb-1"
        >
          <strong>
            ID:
          </strong>
           
          8
        </p>
        <p
          class="mb-1"
        >
          <strong>
            Nombre:
          </strong>
           
          Producto 8
        </p>
        <p
          class="mb-0"
        >
          <strong>
            Ventas semanales:
          </strong>
           
          58
           unidades
        </p>
      </div>
      <div
        class="p-3 text-center"
        style="min-width: 250px; border-radius: 10px; background-color: rgb(253, 250, 249); transition: transform 0.2s ease;"
      >
        <h5
          class="mb-2 text-primary"
        >
          #
          9
        </h5>
        <img
          alt="Producto 9"
          class="d-block mx-auto mb-2"
          src="https://example.com/prod9.jpg"
          style="width: 100px; height: 100px; object-fit: cover;"
        />
        <p
          class="mb-1"
        >
          <strong>
            ID:
          </strong>
           
          9
        </p>
        <p
          class="mb-1"
        >
          <strong>
            Nombre:
          </strong>
           
          Producto 9
        </p>
        <p
          class="mb-0"
        >
          <strong>
            Ventas semanales:
          </strong>
           
          34
           unidades
        </p>
      </div>
      <div
        class="p-3 text-center"
        style="min-width: 250px; border-radius: 10px; background-color: rgb(253, 250, 249); transition: transform 0.2s ease;"
      >
        <h5
          class="mb-2 text-primary"
        >
          #
          10
        </h5>
        <img
          alt="Producto 10"
          class="d-block mx-auto mb-2"
          src="https://example.com/prod10.jpg"
          style="width: 100px; height: 100px; object-fit: cover;"
        />
        <p
          class="mb-1"
        >
          <strong>
            ID:
          </strong>
           
          10
        </p>
        <p
          class="mb-1"
        >
          <strong>
            Nombre:
          </strong>
           
          Producto 10
        </p>
        <p
          class="mb-0"
        >
          <strong>
            Ventas semanales:
          </strong>
           
          27
           unidades
        </p>
      </div>
    </div>
  </div>
</div>
`;
